### ISSUE TITLE
Consider replacing the use of Lambda layers with custom Layer subclasses

### ISSUE THREAD

**Issue by kjun9** [stellargraph/stellargraph#709]
### Description

Motivation is quite clearly stated in the [Tensorflow docs for Lambda layers](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Lambda):
> The Lambda layer exists so that arbitrary TensorFlow functions can be used when constructing Sequential and Functional API models. Lambda layers are best suited for simple operations or quick experimentation. For more advanced usecases, follow this guide for subclassing tf.keras.layers.Layer.

> The main reason to subclass tf.keras.layers.Layer instead of using a Lambda layer is saving and inspecting a Model. Lambda layers are saved by serializing the Python bytecode, whereas subclassed Layers can be saved via overriding their get_config method. Overriding get_config improves the portability of Models. Models that rely on subclassed Layers are also often easier to visualize and reason about.

We currently use `Lambda` layers for doing "simple" things like L2 normalisation or manipulating the matrix dimensions, but these could probably be turned into a set of utility layers we can reuse in different models. 

When inspecting a model via `get_config` a lambda layer typically looks something like (below is a lambda layer used to remove the batch dimension in the adjacency matrix for the GAT model):
```
{'class_name': 'Lambda',
   'config': {'name': 'lambda_1',
    'trainable': True,
    'dtype': 'float32',
    'function': ('4wEAAAAAAAAAAQAAAAQAAABTAAAAcwwAAAB0AKABfABkAaECUwApAk7pAAAAACkC2gFL2gdzcXVl\nZXplKQHaAUGpAHIFAAAA+k0vVXNlcnMvanVuMDIxL3dvcmsvc3RlbGxhci9zdGVsbGFyZ3JhcGgv\nc3RlbGxhcmdyYXBoL2xheWVyL2dyYXBoX2F0dGVudGlvbi5wedoIPGxhbWJkYT5mAwAA8wAAAAA=\n',
     None,
     None),
    'function_type': 'lambda',
    'module': 'stellargraph.layer.graph_attention',
    'output_shape': None,
    'output_shape_type': 'raw',
    'output_shape_module': None,
    'arguments': {}},
   'name': 'lambda_1',
   'inbound_nodes': [[['input_3', 0, 0, {}]]]},
```
which doesn't make it very clear what it's actullay doing.

### User Story

**As** a data scientist

**I want to** use stellargraph models that use custom layer subclasses instead of lambda layers

**So that** my models are easier to inspect and more portable

### Done Checklist

- [ ] Produced code for required functionality
- [ ] Tests written and coverage checked
- [ ] Code review performed
- [ ] Documentation on Google Docs  (if applicable)
- [ ] Documentation in repo
- [ ] Version number reflects new status
- [ ] CHANGELOG.md updated
- [ ] Team demo
